libraries/browser-tracker-core/src/tracker/out_queue.ts

```diff
@@ -1,614 +1,42 @@
-    return !dontRetryStatusCodes.includes(statusCode);
-  }
-
-  /**
-   * Open an XMLHttpRequest for a given endpoint with the correct credentials and header
-   *
-   * @param string - url The destination URL
-   * @returns object The XMLHttpRequest
-   */
-  function initializeXMLHttpRequest(url: string, post: boolean, sync: boolean) {
-    const xhr = new XMLHttpRequest();
-    if (post) {
-      xhr.open('POST', url, !sync);
-      xhr.setRequestHeader('Content-Type', 'application/json; charset=UTF-8');
-    } else {
-      xhr.open('GET', url, !sync);
-    }
-    xhr.withCredentials = withCredentials;
-    if (anonymousTracking) {
-      xhr.setRequestHeader('SP-Anonymous', '*');
-    }
-    for (const header in customHeaders) {
-      if (Object.prototype.hasOwnProperty.call(customHeaders, header)) {
-        xhr.setRequestHeader(header, customHeaders[header]);
-      }
-    }
-    return xhr;
-  }
-
-  /**
-   * Enclose an array of events in a self-describing payload_data JSON string
-   *
-   * @param array - events Batch of events
-   * @returns string payload_data self-describing JSON
-   */
-  function encloseInPayloadDataEnvelope(events: Array<Record<string, unknown>>) {
-    return JSON.stringify({
-      schema: PAYLOAD_DATA_SCHEMA,
-      data: events,
-    });
-  }
-
-  /**
-   * Attaches the STM field to outbound POST events.
-   *
-   * @param events - the events to attach the STM to
-   */
-  function attachStmToEvent(events: Array<Record<string, unknown>>) {
-    const stm = new Date().getTime().toString();
-    for (let i = 0; i < events.length; i++) {
-      events[i]['stm'] = stm;
-    }
-    return events;
-  }
-
-  /**
-   * Creates the full URL for sending the GET request. Will append `stm` if enabled
-   *
-   * @param nextRequest - the query string of the next request
-   */
-  function createGetUrl(nextRequest: string) {
-    if (useStm) {
-      return configCollectorUrl + nextRequest.replace('?', '?stm=' + new Date().getTime() + '&');
-    }
-
-    return configCollectorUrl + nextRequest;
-  }
+  sharedState.bufferFlushers.push(emitter.flush);
 
   return {
-    enqueueRequest: enqueueRequest,
-    executeQueue: () => {
-      if (!executingQueue) {
-        executeQueue();
-      }
-    },
+    enqueueRequest: emitter.input,
+    executeQueue: emitter.flush,
+    setAnonymousTracking: emitter.setAnonymousTracking,
+    setCollectorUrl: emitter.setCollectorUrl,
+    setBufferSize: emitter.setBufferSize,
     setUseLocalStorage: (localStorage: boolean) => {
-      useLocalStorage = localStorage;
-    },
-    setAnonymousTracking: (anonymous: boolean) => {
-      anonymousTracking = anonymous;
-    },
-    setCollectorUrl: (url: string) => {
-      configCollectorUrl = url + path;
-    },
-    setBufferSize: (newBufferSize: number) => {
-      bufferSize = newBufferSize;
+      if (eventStore.hasOwnProperty('setUseLocalStorage')) {
+        const localStorageStore = eventStore as unknown as LocalStorageEventStore;
```

Comment:
Nit: `as LocalStorageEventStore & EventStore` to avoid the `as unknown`.
Alternatively `LocalStorageEventStore` should probably `extends EventStore`.